dotnet/samples/Concepts/Plugins/TransformPlugin.cs

```diff
@@ -0,0 +1,193 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+
+using System.ComponentModel;
+using System.Text.Json.Serialization;
+using Microsoft.SemanticKernel;
+using Microsoft.SemanticKernel.Connectors.OpenAI;
+
+namespace Plugins;
+
+/// <summary>
+/// Sample showing how to transform a <see cref="KernelPlugin"/> so that not all parameters are advertised to the LLM
+/// and instead the argument values are provided by the client code.
+/// </summary>
+public sealed class TransformPlugin(ITestOutputHelper output) : BaseTest(output)
+{
+    /// <summary>
+    /// A plugin that returns favorite information for a user.
+    /// </summary>
+    public class UserFavorites
+    {
+        [KernelFunction]
+        [Description("Returns the favorite color for the user.")]
+        public string GetFavoriteColor([Description("Email address of the user.")] string email)
+        {
+            return email.Equals("bob@contoso.com", StringComparison.OrdinalIgnoreCase) ? "Green" : "Blue";
+        }
+
+        [KernelFunction]
+        [Description("Returns the favorite animal of the specified type for the user.")]
+        public string GetFavoriteAnimal([Description("Email address of the user.")] string email, [Description("Type of animal.")] AnimalType animalType)
+        {
+            if (email.Equals("bob@contoso.com", StringComparison.OrdinalIgnoreCase))
+            {
+                switch (animalType)
```

Comment:
suggest updating to switch expressions to reduce verbosity, for this and the next switch check